Assets/Plugins/Editor/Uplift/UpliftPreferences.cs

```diff
@@ -23,101 +23,82 @@
-            EditorGUILayout.HelpBox(
-                "Experimental features are not thoroughly tested and could induce bugs. Use at your own risk!",
-                MessageType.Warning
-            );
-            useExperimentalFeatures = EditorGUILayout.Toggle("Use experimental features", useExperimentalFeatures);
+		public static string GetGlobalLocation()
+		{
+			string sourceDir = System.IO.Path.Combine(GetHomePath(), folderName);
+			return System.IO.Path.Combine(sourceDir, defaultFileName);
+		}
 
-            EditorGUILayout.LabelField("Github related:", EditorStyles.boldLabel);
-            EditorGUILayout.HelpBox(
-                "Github does not support TLS versions 1.0 and 1.1 as of 22/02/2018, and TLS 1.2 is available in Unity only with Mono 4.8. If you want to use a Github Repository, it will need to be behind a proxy",
-                MessageType.Info
-            );
-            useGithubProxy = EditorGUILayout.Toggle("Use Github proxy", useGithubProxy);
-            GUI.enabled = useGithubProxy;
-            if (useGithubProxy)
-                EditorGUILayout.HelpBox(
-                    "The proxy URL will replace 'https://api.github.com'",
-                    MessageType.Info
-                );
-            githubProxyUrl = EditorGUILayout.TextField("Proxy URL", githubProxyUrl);
-            githubProxyUrl = Regex.Replace(githubProxyUrl, "/$", "");
-            GUI.enabled = true;
+		public static string GetProjectLocation()
+		{
+			throw new NotImplementedException();
+		}
 
-            if (GUI.changed)
-            {
-                EditorPrefs.SetBool(useExperimentalFeaturesKey, useExperimentalFeatures);
-                EditorPrefs.SetBool(trustUnknownCertificatesKey, trustUnknownCertificates);
-                EditorPrefs.SetBool(githubProxyUseKey, useGithubProxy);
-                EditorPrefs.SetString(githubProxyUrlKey, githubProxyUrl);
-            }
-        }
+		public static bool TryLoadFromFile(string source, out UpliftPreferences result)
+		{
+			result = new UpliftPreferences();
 
-        private static void EnsurePrefsLoaded()
-        {
-            if (!prefsLoaded)
-            {
-                useExperimentalFeatures = EditorPrefs.GetBool(useExperimentalFeaturesKey, false);
-                trustUnknownCertificates = EditorPrefs.GetBool(trustUnknownCertificatesKey, false);
-                useGithubProxy = EditorPrefs.GetBool(githubProxyUseKey, false);
-                githubProxyUrl = EditorPrefs.GetString(githubProxyUrlKey, "");
-                prefsLoaded = true;
-            }
-        }
+			if (!File.Exists(source))
+			{
+				Debug.Log("No preferences file at " + source);
+				return false;
+			}
 
-        public static bool UseExperimental()
-        {
-            EnsurePrefsLoaded();
-            return EditorPrefs.GetBool(useExperimentalFeaturesKey, false);
-        }
+			StrictXmlDeserializer<UpliftPreferences> deserializer = new StrictXmlDeserializer<UpliftPreferences>();
```

Comment:
Do we want to be strict again? Won't that prevent smooth upgrades?

Reply:
No you're right we don't. This is now fixed